Pegasus/include/pegasus/Material.hpp

```diff
@@ -45,47 +45,50 @@ struct Material
      * @brief Checks if the body has an infinite mass
      * @return @c true if the mass is infinite, @c false otherwise
      */
-    bool HasInfiniteMass() const;
+    inline bool HasInfiniteMass() const
+    {
+        return m_inverseMass == 0;
```

Comment:
```suggestion
        return std::isnan(m_inverseMass);
```

Not completely sure about this one. Would it work with `-0.0f` as well as with `0.0f`? Or are we setting `0` explicitly?
Can we use some other value like `NAN` for infinite mass?

Reply:
+

Reply:
We can. But. Using zero has its advantages, for example, I can use it in the equations when there is a proportion involved and all the momentum will be transferred to the body with non zero mass. This allows me to avoid unnecessary brunching, so far it's been working so I don't see any particular problem that we have now that could be solved with this change.

Reply:
And we do set it explicitly.